addons/sale_coupon/models/sale_order.py

```diff
@@ -147,10 +152,11 @@ def _get_reward_values_discount(self, program):
             }]
         reward_dict = {}
         lines = self._get_paid_order_lines()
+        amount_total = sum(self._get_base_order_lines(program).mapped('price_total'))
```

Comment:
This should be `price_subtotal`.
Depending on the promotion order between the 2, result might be still negative.

### Using `price_total`
Fixed amount first:
![image](https://user-images.githubusercontent.com/30048408/104332370-65b24c00-54f0-11eb-8f41-b2a72571689b.png)

Discount first:
![image](https://user-images.githubusercontent.com/30048408/104332605-a27e4300-54f0-11eb-83eb-252c19f99d68.png)



### Using `price_subtotal`, same flows will become:
Fixed amount first:
![image](https://user-images.githubusercontent.com/30048408/104332845-e7a27500-54f0-11eb-9db7-cf14fa8aea51.png)


Discount first:
![image](https://user-images.githubusercontent.com/30048408/104333150-33551e80-54f1-11eb-810b-8ad587ec634f.png)